### PR TITLE
feat: Added CI Tests

### PR DESCRIPTION
**Pull request by samyakjain10**

### CODE REVIEW

**Comment by programmer290399** [comment]
@samyakjain10 Please also add a GitHub workflow in a .github folder as done in cluster-stats

**Comment by samyakjain10** [comment]
@programmer290399 Already done that in [43cfad0](https://github.com/CRED-CLUB/kong-plugin-qos-classifier/pull/6/commits/43cfad042233d88186b757d74b750a8f3fc56a34)

File: spec/qos-classifier/02-integration_spec.lua [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      assert(bp.plugins:insert{
+        name = PLUGIN_NAME,
+        route = route4,
+        config = config 
+      })
+
+      -- Start kong
+      assert(helpers.start_kong({
+        database   = strategy,
+        nginx_conf = "/kong-plugin/spec/fixtures/custom_nginx.template",
```
Comment (Abhishekvrshny) [inline spec/qos-classifier/02-integration_spec.lua:114]: why is this custom nginx template required?

File: spec/qos-classifier/02-integration_spec.lua [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  upstream_header_name = "X-QOS-CLASS",
+  node_count = {
+      http_timeout_in_ms = 15,
+      update_initial_delay_in_sec = 5,
+      initial = 1,
+      update_frequency_in_sec = 1
+  }
+}
+
+local function make_class(green, orange, red)
```
Comment (Abhishekvrshny) [inline spec/qos-classifier/02-integration_spec.lua:21]: nit: can be renamed to `set_classes_with_thresholds` to make it more intuitive

File: spec/qos-classifier/02-integration_spec.lua [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+local function make_requests_to_upstream(host, start_time)
+  local client
+  while true do
+    client = helpers.proxy_client()
+    assert(client:get("/get", {
+      headers = { Host = host },
+    }))
+    ngx.sleep(0.01)
+    if(ngx.now() - start_time > 1.2) then break end
```
Comment (Abhishekvrshny) [inline spec/qos-classifier/02-integration_spec.lua:47]: A comment on why 1.2 would help here

File: spec/qos-classifier/02-integration_spec.lua [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+for _, strategy in helpers.each_strategy() do
+  describe(PLUGIN_NAME .. ": (integration) [#" .. strategy .. "]", function()
+    local bp, db
+  
+    lazy_setup(function()
+      bp, db = helpers.get_db_utils(strategy, nil, { PLUGIN_NAME })
+      local service = bp.services:insert()
+
+      local route1 = bp.routes:insert {
+        hosts   = {  "test1.com" },
```
Comment (Abhishekvrshny) [inline spec/qos-classifier/02-integration_spec.lua:60]: "test1.com", "test2.com" etc can be put in a variable for reusability and can have more intuitive values like "terminate.host", "green.host" etc

File: spec/qos-classifier/02-integration_spec.lua (hunk 1 above)
Comment (samyakjain10) [reply spec/qos-classifier/02-integration_spec.lua:114]: Custom nginx template is required to add `shared_dict` s to nginx configuration which are required by our plugin. Check [81726e1](https://github.com/CRED-CLUB/kong-plugin-qos-classifier/pull/6/commits/81726e13dcc9e18b7e41f61b78de71606e9ea2a1)

File: spec/qos-classifier/02-integration_spec.lua (hunk 4 above)
Comment (samyakjain10) [reply spec/qos-classifier/02-integration_spec.lua]: Done

File: spec/qos-classifier/02-integration_spec.lua (hunk 2 above)
Comment (samyakjain10) [reply spec/qos-classifier/02-integration_spec.lua]: Done

File: spec/qos-classifier/02-integration_spec.lua (hunk 3 above)
Comment (samyakjain10) [reply spec/qos-classifier/02-integration_spec.lua:61]: Tried my best to comment the logic down in  [4129758](https://github.com/CRED-CLUB/kong-plugin-qos-classifier/pull/6/commits/4129758beee8e3033a53e32000cd45780c74b130). Please review.

**Review by Abhishekvrshny — APPROVED** [review]

**Pull request merged**